examples/caching/cdi/src/main/java/io/helidon/examples/caching/cdi/CachedBean.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.examples.caching.cdi;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.helidon.caching.Cache;
+import io.helidon.caching.annotation.CacheGet;
+import io.helidon.caching.annotation.CacheInit;
+import io.helidon.caching.annotation.CacheKey;
+import io.helidon.caching.annotation.CacheName;
+import io.helidon.caching.annotation.CachePut;
+import io.helidon.caching.annotation.CacheRemove;
+import io.helidon.caching.annotation.CacheValue;
+
+@ApplicationScoped
+@CacheName("simple-cache")
+public class CachedBean {
+    private final AtomicInteger counter = new AtomicInteger();
+    private final Cache<Integer, Integer> cache;
+
+    @Inject
+    CachedBean(@CacheName("loader-cache") Cache<Integer, Integer> cache) {
+        this.cache = cache;
+        System.out.println(cache.get(79).await());
+    }
+
+    @CacheGet
+    public String getIt(@CacheKey Integer id) {
+        return "produced-" + id + "-" + counter.incrementAndGet();
+    }
+
+    @CacheInit
+    public String createIt(@CacheKey Integer id) {
```

Comment:
How is `@CacheInit` different from `@CacheGet`? Is the idea to initialize the cache with a response, but always execute the method, where `@CacheGet` would only execute the method if the value is not cached?

Reply:
Yes, that is the distinction